src/conn/conn_sweep.c

```diff
@@ -261,6 +261,67 @@ __sweep_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_SWEEP));
 }
 
+/*
+ * __sweep_check_session_sweep --
+ *     Check for any "rogue" sessions, which did not run a session sweep in a long time.
+ */
+static void
+__sweep_check_session_sweep(WT_SESSION_IMPL *session, uint64_t now)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_SESSION_IMPL *s;
+    uint64_t last, last_cursor_big_sweep, last_sweep;
+    uint32_t i;
+
+    conn = S2C(session);
+
+    for (s = conn->sessions, i = 0; i < conn->session_cnt; ++s, ++i) {
+        /*
+         * Ignore inactive and internal sessions.
+         */
+        if (!s->active)
+            continue;
+        if (F_ISSET(s, WT_SESSION_INTERNAL))
+            continue;
+
+        last_cursor_big_sweep = s->last_cursor_big_sweep;
+        last_sweep = s->last_sweep;
+
+        last = last_cursor_big_sweep;
+        if (last_sweep > last)
```

Comment:
I wonder if want this inequality reversed?  E.g. last_cursor_big_sweep is timestamp 10000 and the last_sweep is 50000 and now is 51000.  Maybe last_sweep is keeping up, but the cursor sweep is not.  It doesn't look like this catches that?  Or am I missing something?

These two sweeps clean up two data structures.  Both of the structures separately may hold references to dhandles.

Reply:
I didn't realize that these were two different processes, in which case we should then take the earlier non-zero timestamp.